rotkehlchen/exchanges/ftx.py

```diff
@@ -309,15 +334,25 @@ def _api_query(
     @protect_with_lock()
     @cache_response_timewise()
     def query_balances(self) -> ExchangeQueryBalances:
+        resp_dict, resp_lst = None, None
         try:
-            resp = self._api_query('wallet/all_balances', paginate=False)
+            if self.subaccount is not None:
+                resp_lst = self._api_query('wallet/balances', paginate=False)
+            else:
+                resp_dict = self._api_query('wallet/all_balances', paginate=False)
         except RemoteError as e:
             msg = f'FTX API request failed. Could not reach FTX due to {str(e)}'
             log.error(msg)
             return None, msg
 
-        # flatten the list that maps accounts to balances
-        balances = [x for _, bal in resp.items() for x in bal]
+        if resp_dict is not None:
+            # flatten the list that maps accounts to balances
+            balances: List[Dict[str, Any]] = [x for _, bal in resp_dict.items() for x in bal]
+        elif resp_lst is not None:
+            # When querying for a subaccount we get a list instead of a dict
+            balances = resp_lst
+        else:
+            return None, 'Couldnt retrieve a valid response from server'
```

Comment:
```suggestion
            return None, 'Could not retrieve a valid response from server'
```

This final else is not needed. It can never happen. Figure out a better way to deal with the types here and don't add extra code that's not needed.